packages/state-transition/src/block/slashValidator.ts

```diff
@@ -37,9 +37,12 @@ export function slashValidator(
   validator.withdrawableEpoch = Math.max(validator.withdrawableEpoch, epoch + EPOCHS_PER_SLASHINGS_VECTOR);
 
   const {effectiveBalance} = validator;
-  // TODO: could state.slashings be number?
+  // state.slashings is a number because it's reset per epoch in processSlashingsReset()
+  // for each epoch, there are 8704 max validators to slash so it's safe to use Number
+  // also we don't need to compute the total slashings from state.slashings, it's handled by totalSlashingsByIncrement in EpochCache
   const slashingIndex = epoch % EPOCHS_PER_SLASHINGS_VECTOR;
-  state.slashings.set(slashingIndex, state.slashings.get(slashingIndex) + BigInt(effectiveBalance));
+  state.slashings.set(slashingIndex, state.slashings.get(slashingIndex) + effectiveBalance);
+  epochCtx.totalSlashingsByIncrement += effectiveBalanceIncrements[slashedIndex];
```

Comment:
can `effectiveBalanceIncrements[slashedIndex]` be undefined? may be worth adding `?? 0`

Reply:
`effectiveBalanceIncrements` is populated with validator count and increased its length per `processDeposit` so the default value for any validator index is always 0

I also doubled check all use of `effectiveBalanceIncrements`, we don't need to handle `undefined` there so I think we don't need to do it here too. For example https://github.com/ChainSafe/lodestar/blob/3f3a207643f6c35ae4b3cfe683cd167cfe70a2c3/packages/state-transition/src/block/processAttestationsAltair.ts#L111